arcgis-tiledlayer-rendermode/src/main/java/com/esri/arcgisruntime/sample/arcgistiledlayerrendermode/MainActivity.java

```diff
@@ -0,0 +1,311 @@
+import com.esri.arcgisruntime.sample.android.widget.VerticalSeekBar;
+
+public class MainActivity extends AppCompatActivity {
+
+  private MapView mMapView;
+  private List<String> mRendermodeSpinnerList;
+  private ArrayAdapter<String> mDataAdapter;
+  private VerticalSeekBar mSeekBar;
+  private TextView mSeekBarScale;
+  private TextView mMapViewScale;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    // inflate MapView from layout
+    mMapView = (MapView) findViewById(R.id.mapView);
+
+    final ArcGISTiledLayer tiledLayerBaseMap = new ArcGISTiledLayer(getResources().getString(R.string.world_topo_service));
+
+    // set tiled layer as basemap
+    Basemap basemap = new Basemap(tiledLayerBaseMap);
+    // create a map with the basemap
+    Map map = new Map(basemap);
+    // set the map to be displayed in this view
+    mMapView.setMap(map);
+
+    // populate the spinner list with default bookmark names
+    Spinner rendermodesSpinner = (Spinner) findViewById(R.id.rendermodesspinner);
+    mRendermodeSpinnerList = new ArrayList<>();
+    mRendermodeSpinnerList.add("RenderMode - DEFAULT");
+    mRendermodeSpinnerList.add("RenderMode - AESTHETIC");
+    mRendermodeSpinnerList.add("RenderMode - SCALE");
+
+    // initialize the adapter for the bookmarks spinner
+    mDataAdapter = new ArrayAdapter<>(this,
+        R.layout.spinner_item, mRendermodeSpinnerList);
+    rendermodesSpinner.setAdapter(mDataAdapter);
+
+    // when an item is selected in the spinner set the mapview viewpoint to the selected
+    // bookmark's viewpoint
+    rendermodesSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+      @Override
+      public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        switch (position) {
+          case 0 :
+            tiledLayerBaseMap.setRenderMode(ImageTiledLayer.RenderMode.DEFAULT);
+            break;
+          case 1 :
+            tiledLayerBaseMap.setRenderMode(ImageTiledLayer.RenderMode.AESTHETIC);
+            break;
+          case 2 :
+            tiledLayerBaseMap.setRenderMode(ImageTiledLayer.RenderMode.SCALE);
+            break;
+
+        }
+
+      }
+      @Override
+      public void onNothingSelected(AdapterView<?> parent) {
+      }
+    });
+
+    mSeekBar = (VerticalSeekBar) findViewById(R.id.seekBar);
+    mSeekBarScale = (TextView) findViewById(R.id.seekbarscale);
+    mMapViewScale = (TextView) findViewById(R.id.mapviewscale);
+
+    mSeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+      @Override public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+        switch (progress) {
+          case 0:
+            mMapView.setViewpointScaleAsync(5.91657527591555E8);
+            mSeekBarScale.setText("Set Scale - "+ 5.91657527591555E8);
```

Comment:
This applies to all the `setText()` methods below:  Can be simplified using string resources with placeholders instead of concatenating strings displayed with `setText()`.  

String resource:
```xml
<string name="seekbar_scale">Seekbar Scale - %1$d</string>
```

usage
```java
getString(R.string.seekbar_scale, 5.91657527591555E8);
```